### PR TITLE
Add just-based lint/test workflow and UPS status refinements

### PR DESCRIPTION
**Pull request by viell-dev**
## Summary
- Add a `just` command surface for local lint, test, and combined check runs, and switch CI to use those shared recipes.
- Expand the UPS script behavior to support configurable charge thresholds, better error handling, and more complete JSON escaping.
- Update the README and changelog to reflect the new install flow, status semantics, and testing commands.
- Add coverage for the new threshold logic, malformed output handling, and escaping edge cases.

## Testing
- `just lint`
- `just test`
- CI runs the same shared lint and test commands via GitHub Actions.

## Summary by CodeRabbit

* **New Features**
  * Configurable UPS charge warning/critical thresholds; improved error/tooltip reporting and JSON-safe output.

* **Bug Fixes**
  * More robust status parsing, clearer severity classification (warning vs critical), and safer escaping for displayed text.

* **Documentation**
  * Updated installation, usage, testing, and troubleshooting guidance; consolidated behavioral notes to a single source.

* **Tests**
  * Expanded tests for thresholds, status edge cases, and error paths.

* **Chores**
  * Added task automation and CI tasks to standardize linting and test runs.

[Review Change Stack](https://app.coderabbit.ai/change-stack/viell-dev/waybar-apcupsd/pull/6?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: ups-status.sh [hunk 1]
```diff
@@ -262,22 +371,48 @@ fi
...
 # --- Build output ---------------------------------------------------------
 
-TEXT="${FALLBACK_TEXT:-${CHARGE_INT}%}"
+if [[ -n "$FALLBACK_TEXT" ]]; then
+    TEXT=$FALLBACK_TEXT
+elif (( CHARGE_VALID == 1 )); then
+    TEXT="${CHARGE_INT}%"
+else
+    TEXT="$TEXT_UNKNOWN"
+fi
```
[the remark this answers is not in the document]
Comment (viell-dev) [reply ups-status.sh:380]: stale, no action needed
